File: backend/app/api/v1/routes/schemas.py
from uuid import UUID

from fastapi import APIRouter, Depends, status

from app.core.security import AuthenticatedUser, get_current_user
from app.models.requests.schema import CreateSchemaRequest, UpdateSchemaRequest
from app.models.responses.schema import SchemaResponse, SchemaListResponse

router = APIRouter()


@router.get("", response_model=SchemaListResponse)
async def list_schemas(
    project_id: UUID,
    cursor: str | None = None,
    limit: int = 20,
    current_user: AuthenticatedUser = Depends(get_current_user),
) -> SchemaListResponse:
    raise NotImplementedError


@router.post("", response_model=SchemaResponse, status_code=status.HTTP_201_CREATED)
async def create_schema(
    body: CreateSchemaRequest,
    current_user: AuthenticatedUser = Depends(get_current_user),
) -> SchemaResponse:
    raise NotImplementedError


@router.get("/{schema_id}", response_model=SchemaResponse)
async def get_schema(
    schema_id: UUID,
    current_user: AuthenticatedUser = Depends(get_current_user),
) -> SchemaResponse:
    raise NotImplementedError


@router.patch("/{schema_id}", response_model=SchemaResponse)
async def update_schema(
    schema_id: UUID,
    body: UpdateSchemaRequest,
    current_user: AuthenticatedUser = Depends(get_current_user),
) -> SchemaResponse:
    raise NotImplementedError


@router.delete("/{schema_id}", status_code=status.HTTP_204_NO_CONTENT)
async def delete_schema(
    schema_id: UUID,
    current_user: AuthenticatedUser = Depends(get_current_user),
) -> None:
    raise NotImplementedError

File: frontend/lib/api/schemas.ts
import { apiFetch } from "@/lib/api/client";
import type { Schema, SchemaListResponse, CreateSchemaInput } from "@/types/schema";

export async function listSchemas(projectId: string, cursor?: string): Promise<SchemaListResponse> {
  const params = new URLSearchParams({ project_id: projectId });
  if (cursor) params.set("cursor", cursor);
  return apiFetch<SchemaListResponse>(`/schemas?${params}`);
}

export async function getSchema(schemaId: string): Promise<Schema> {
  return apiFetch<Schema>(`/schemas/${schemaId}`);
}

export async function createSchema(input: CreateSchemaInput & { project_id: string }): Promise<Schema> {
  return apiFetch<Schema>("/schemas", { method: "POST", body: JSON.stringify(input) });
}

export async function deleteSchema(schemaId: string): Promise<void> {
  return apiFetch<void>(`/schemas/${schemaId}`, { method: "DELETE" });
}
